delta/src/main/scala/io/qbeast/spark/delta/DeltaMetadataWriter.scala

```diff
@@ -149,11 +152,13 @@ private[delta] case class DeltaMetadataWriter(
    * @return
    *   A Map[String, String] representing the combined outputs of all hooks.
    */
-  private def runPreCommitHooks(actions: Seq[Action]): PreCommitHookOutput = {
-    preCommitHooks.foldLeft(Map.empty[String, String]) { (acc, hook) => acc ++ hook.run(actions) }
+  private def runPreCommitHooks(actions: Seq[QbeastFile]): PreCommitHookOutput = {
+    preCommitHooks.foldLeft(Map.empty[String, String]) { (acc, hook) =>
+      acc ++ hook.run(actions)
+    }
   }
 
-  def writeWithTransaction(writer: => (TableChanges, Seq[FileAction])): Unit = {
+  def writeWithTransaction(writer: => (TableChanges, Seq[IndexFile], Seq[DeleteFile])): Unit = {
```

Comment:
Again, maybe not for this PR, but could we create our qbeast `FileAction` abstraction for the action committed on the transaction? Or is it really useful to have Index and Deleted files in separated variables?

Reply:
That way, we can have a single map for both sequences and in the future we could add different actions to the list. 

Up to discussion. what do you think?

Reply:
Opened issue #449